### ISSUE TITLE
Problem with training dataset creation after label refinements

### ISSUE THREAD

**Issue by EliaFe** [DeepLabCut/DeepLabCut#656]
**Your Operating system and DeepLabCut version**
Windows 10, Anaconda environement, python 3.6.9, Deeplabcut 2.1.6.4
Browser: chrome

**Describe the problem**
Hi,
I wanted to refine the model, hence I extracted the outliers, refined the labels and merged the dataset. This error (see screenshot) appears when I want to create a new training dataset with these refinements.
I tried to search the cause of this error but I am stuck. From what I have search, it appears that the folder "iteration-1" is created in the "training-datasets", with the files .csv and h5 files but the matlab file can't be created.
Also I think (correct me if I am wrong) that the initial and refined frames are merged since there is 16 frames per videos and the initial labelling is 8.

Many thanks for your help.

**How to Reproduce the problem**
Steps to reproduce the behavior: (not sure that this will reproduce the error though because I did the same thing for other model refinements and all were okay)
1. Outliers extraction
2. Refinement
3. Dataset merged
4. New training set creation with mobilnet_v2_1.0, imgaug augmentation method

**Screenshots**

![trainset_creation2](https://user-images.githubusercontent.com/22397038/79611532-cc711900-80c8-11ea-9187-1b9237b0e6ea.png)

**Comment by AlexEMG** [comment]
I would first run `deeplabcut.check_labels` to see if the labels are all there (i.e. all folders contain h5 files etc.). I suppose some data is missing/corrupt?

**Comment by MMathisLab** [comment]
> Also I think (correct me if I am wrong) that the initial and refined frames are merged since there is 16 frames per videos and the initial labelling is 8.


Can you confirm you re-labeled in an already labeled video, vs. on a novel video?

**Comment by EliaFe** [comment]
I have run deeplabcut.check_labels. There is no error appearing and the folders containing the labelled images for each videos are created. However, all labels are nice except one. It is like the label coordinates were the ones from another frame (some look like the last frame of the previous video but not always). It seems that this question has already been mentionned in  #574 and #581 (but I am using only windows plateform).
I have checked if it was doing the same for another model and it is also the case.

@MMathisLab I am not sure to understand your question.

**Comment by AlexEMG** [comment]
Ok, I suppose if you fix that image, then the training works?

**Comment by EliaFe** [comment]
No, I reviewed all the images and I had the same error.
But I found the problem, apparently the first 3 frames of the first video had 2 additional labels that are not appearing on the config file. If I exclude this video to do the training set, it works.
Thanks for you help !

**Comment by AlexEMG** [comment]
You can just use the normal labeling GUI to fix this one frame and you are good to train!

________________________________

**Comment by EliaFe** [comment]
What I don't really understand is that this label does not appear on the GUI but it is written in the excel file...

**Comment by AlexEMG** [comment]
I see, is the image in the folder?

________________________________

**Comment by EliaFe** [comment]
The images are in the folder, but they are not labelled after the _check_labels_  since these are the initial ones and not the refined.

**Comment by AlexEMG** [comment]
That’s not how it should work. Check-labels should run over all folders and their Images

________________________________

**Comment by EliaFe** [comment]
Sorry you're right. The additional label present in the excel (with coordinates written) is not present on the labelled photos.

**Comment by MMathisLab** [comment]
so then you can run: `deeplabcut.dropannotationfileentriesduetodeletedimages` see: https://github.com/AlexEMG/DeepLabCut/blob/master/docs/HelperFunctions.md and the doccstring 

`deeplabcut.dropannotationfileentriesduetodeletedimages?`

[ of course use with caution; perhaps make a back-up copy of your labeled-data folder first]

**Comment by EliaFe** [comment]
Great, I have run this. Thanks.
